protocol/src/protocol.ts

```diff
@@ -1207,6 +1207,26 @@ export namespace ShowMessageRequest {
 	export const type = new RequestType<ShowMessageRequestParams, MessageActionItem | null, void, void>('window/showMessageRequest');
 }
 
+export interface ShowTextDocumentRequestParams {
+	/**
+	 * A name of a file on disk.
+	 */
+	fileName: string;
+
+	/**
+	 * Editor options to configure the behavior of showing the editor.
+	 */
+	options: TextDocumentShowOptions;
```

Comment:
As all the properties within `TextDocumentShowOptions` is optional, should `options` itself be `options?` instead?

Reply:
It's optional now, thanks.